ThunkLibs/libvulkan/Host.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+$info$
+tags: thunklibs|Vulkan
+$end_info$
+*/
+
+#define VK_USE_PLATFORM_XLIB_XRANDR_EXT
+#define VK_USE_PLATFORM_XLIB_KHR
+#define VK_USE_PLATFORM_XCB_KHR
+#define VK_USE_PLATFORM_WAYLAND_KHR
+#include <vulkan/vulkan.h>
+
+#include "common/Host.h"
+
+#include <cstring>
+#include <mutex>
+#include <unordered_map>
+
+#include <dlfcn.h>
+
+#include "ldr_ptrs.inl"
+
+static bool SetupInstance{};
+static std::mutex SetupMutex{};
+
+#define LDR_PTR(fn) fexldr_ptr_libvulkan_##fn
+
+static void DoSetupWithInstance(VkInstance instance) {
```

Comment:
Is this still needed with GCH?

From what I understand we can remove special handling for all of these

Reply:
This relies on https://github.com/FEX-Emu/FEX/issues/1774.

Until the GCH framework supports functions with custom guest entrypoints, vkGetDeviceProcAddr needs `fexldr_ptr_vkGetDeviceProcAddr` to be set up before being called (which in turn needs a VkInstance to be present).